lib/src/main/java/org/esbtools/eventhandler/NotificationRepository.java

```diff
@@ -21,16 +21,35 @@
 import java.util.Collection;
 import java.util.List;
 
+/**
+ * Abstracts a transactional backing store and staging area for {@link Notification notifications}.
+ *
+ * <p>A repository is responsible for handling CRUD and data parsing operations around these
+ * objects, in whatever scheme necessary.
+ *
+ * <p>Production implementations are expected to be thread safe, even across a network.
+ */
 public interface NotificationRepository {
 
     /**
-     * Retrieves the oldest {@code maxEvents} {@link Notification notifications}, oldest first.
+     * Retrieves the oldest {@code maxNotifications} {@link Notification notifications}, oldest
+     * first.
      *
      * <p>Once retrieved, a notification should not be retrieved again, atomically. That is, many
-     * threads looking up notifications at the same time should all et a unique non-overlapping sample
-     * of the oldest notifications. Subsequent calls should always return a unique set.
+     * threads looking up notifications at the same time should all et a unique non-overlapping
+     * sample of the oldest notifications. Subsequent calls should always return a unique set.
```

Comment:
Do you think it would be an excessive amount of lock calls to lock for each notification id? Notification batches will typically be large I think; we should process notifications as fast as lightblue will allow us.